online/src/app/product-details/product-details.component.ts

```diff
@@ -14,31 +16,34 @@ import { admin } from '../utils';
   styleUrls: ['./product-details.component.scss']
 })
 export class ProductDetailsComponent implements OnInit {
-  product: Product | undefined;
+  id: number | undefined;
   products: ProductViewModel[] | undefined;
-  hasAdminRole = this.authService.hasRoleType(admin);
+  roleSelector = this.store.select(adminRoleSelector);
+  hasAdminRole: boolean | undefined;
+  product: Product | undefined;
+  selectedProduct$ = this.store.select(selectOneProduct);
 
-  constructor(private productService: ProductService, private location: Location, private route: ActivatedRoute, private authService: AuthentificationService) {
 
-  }
+  constructor(private location: Location,
+    private route: ActivatedRoute,
+    private store: Store<AppState>) {
 
-  ngOnInit(): void {
-    const id = this.route.snapshot.params['id'];
-    this.getProductById(id).subscribe((item: Product) => this.product = item);
   }
 
-  getProductById(id: number): Observable<Product> {
-    return this.productService.getProductById(id);
+  ngOnInit(): void {
+    this.id = this.route.snapshot.params['id'];
+    if (this.id) {
+      this.store.dispatch(getProduct({ id: this.id }));
+    }
+    this.roleSelector.subscribe((role) => { this.hasAdminRole = role });
   }
 
-  deleteProduct(id: number): void {
-    this.productService.deleteProduct(id).subscribe(() => {
-      this.productService.getProducts().subscribe((products) => {
-        this.products = products;
-        alert(`The product with ID ${id} has been successfully deleted`);
-        this.goBack();
-      });
-    });
+  deleteProduct(): void {
+    if (this.id) {
+      this.store.dispatch(removeProduct({ id: this.id }))
+      alert(`The product with ID ${this.id} has been successfully deleted`);
```

Comment:
Alerts such as this, as well as returning, I would generally put in the effects. In a real world application, you'd have the alert extracted in a separate thing which you can just inject into the effect class, just as you can with the router.